src/seedu/addressbook/Main.java

```diff
@@ -105,13 +110,17 @@ private void recordResult(CommandResult result) {
      * @param command user command
      * @return result of the command
      */
-    private CommandResult executeCommand(Command command)  {
+    private CommandResult executeCommand(Command command) throws StorageOperationException {
         try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
             storage.save(addressBook);
             return result;
-        } catch (Exception e) {
+        } catch (StorageOperationException soe){
+            ui.showToUser(Messages.MESSAGE_READ_ONLY);
+            throw new StorageOperationException(Messages.MESSAGE_READ_ONLY);
```

Comment:
If we throw the exception here, someone upstream will have to catch it. Since we have already handled it (informing the user), we don't need to throw it again.